### PR TITLE
Use the proc filesystem to gather process information

### PR DESCRIPTION
**Pull request by nmandery**
## Description
This pull request uses the proc filesystem to gather process information instead of calling the `ps` command.

## Motivation and Context
See the discussion at PR #1876

### CODE REVIEW

**Comment by erikbern** [comment]
cool

File: luigi/lock.py [hunk 1]
```diff
@@ -42,14 +42,17 @@ def getpcmd(pid):
...
+        # Use the /proc filesystem
+        # At least on android there have been some issues with not all
+        # process infos being readable. In these cases using the `ps` command
+        # worked. See the pull request at
+        # https://github.com/spotify/luigi/pull/1876
+        try:
+            with open('/proc/{0}/cmdline'.format(pid), 'r') as fh:
+                return fh.read().replace('\0', ' ').rstrip()
+        except IOError:
+            pass
```
Comment (nmandery) [inline luigi/lock.py]: I just added commit 1836dd2 with a comment

**Comment by Tarrasch** [comment]
Isn't it better to actually throw an exception instead of using a fallback-method when there's a file permissions error? I imagine an admin mis-configuring so that two different users run the same command. It's better to report an error early rather than always running in a `--no-lock` fashion. Have I misunderstood anything?

**Comment by nmandery** [comment]
The former implementation as well as the implementation for windows systems both do not raise any errors when a pid can not be resolved. Reasons I can think of may be missing permissions or simply that the process in the meantime has exited. Instead the fallback at the end of the function is used. So there is always a string retruned giving some description on the process.

This PR simply preserves this behavior of the getpcmp function. The function tries to resolve the command of the given pid and returns the most meaningful description possible.

In case the desired behavior of `getpcmd` has changed to raise exceptions when resolving the command fails, I do not see it as sufficient to propagate the IOError from the failing access to `/proc`. The windows implementation would also be in need of a patch to also raise an exception when the pid can not be found in the output of the `wmic` command. This patch would then result in making the fallback at the end of the function non-reachable and obsolete.

Maybe you should also discuss this with @erikbern , the original author of the `getpcmd` function

**Comment by Tarrasch** [comment]
@nmandery, You seem to have thought about this more thoroughly than I have. And admittedly know more about OS/procs/etc than me. :) Furthermore you just documented your thought-process in case further committers want to read up on this. Thanks. I feel confident enough to merge this. :)

**Comment by Tarrasch** [comment]
@nmandery, I took the liberty to myself summarize why we did this in the squash-commit I just merged this PR with. I hope it looks good.

**Comment by nmandery** [comment]
@Tarrasch Looks fine to me. Thank you for merging and the reviews.

**Pull request merged**